packages/vm/src/evm/opcodes/functions.ts

```diff
@@ -1023,33 +1024,56 @@ export const handlers: Map<number, OpHandler> = new Map([
   [
     0xf6,
     async function (runState) {
-      const [commitUnpadded, yParity, r, s] = runState.stack.popN(4)
-      if (s > SECP256K1_ORDER_DIV_2) {
+      // eslint-disable-next-line prefer-const
+      let [authority, memOffset, memLength] = runState.stack.popN(3)
+
+      if (memLength > BigInt(128)) {
+        memLength = BigInt(128)
+      }
+
+      let mem = runState.memory.read(Number(memOffset), Number(memLength))
+      if (mem.length < 128) {
+        mem = setLengthRight(mem, 128)
+      }
+
+      const yParity = BigInt(mem[31])
```

Comment:
Hm, I don't think this line is correct. EIP-3074 says that:

> `memory[offset : offset+32 ]` - `yParity`

So `yParity` could, in theory, be larger than one byte. As specified, signature verification would fail in that case, but with this implementation, it looks like the higher bytes are ignored?

Reply:
@jochem-brouwer could you have a look?